config/include.go

```diff
@@ -470,3 +501,203 @@ func mergeHooks(terragruntOptions *options.TerragruntOptions, childHooks []Hook,
+// encoded. That is, all of the following are valid encodings of a terragrunt.hcl.json file that has a bare include
+// block:
+//
+// Case 1: a single include block as top level:
+// {
+//   "include": {
+//     "path": "foo"
+//   }
+// }
+//
+// Case 2: a single include block in list:
+// {
+//   "include": [
+//     {"path": "foo"}
+//   ]
+// }
+//
+// Case 3: mixed bare and labeled include block as list:
+// {
+//   "include": [
+//     {"path": "foo"},
+//     {
+//       "labeled": {"path": "bar"}
+//     }
+//   ]
+// }
+//
+// For simplicity of implementation, we focus on handling Case 1 and 2, and ignore Case 3. If we see Case 3, we will
+// error out. Instead, the user should handle this case explicitly using the object encoding instead of list encoding:
+// {
+//   "include": {
+//     "": {"path": "foo"},
+//     "labeled": {"path": "bar"}
+//   }
+// }
+// If the multiple include blocks are encoded in this way in the json configuration, nothing needs to be done by this
+// function.
+func updateBareIncludeBlockJSON(fileBytes []byte) ([]byte, bool, error) {
+	var parsed map[string]interface{}
+	if err := json.Unmarshal(fileBytes, &parsed); err != nil {
+		return nil, false, errors.WithStackTrace(err)
+	}
+	includeBlock, hasKey := parsed["include"]
+	if !hasKey {
+		// No include block, so don't do anything
+		return fileBytes, false, nil
+	}
+	switch typed := includeBlock.(type) {
+	case []interface{}:
+		if len(typed) == 0 {
+			// No include block, so don't do anything
+			return nil, false, nil
+		} else if len(typed) > 1 {
+			// Could be multiple bare includes, or Case 3. We simplify the handling of this case by erroring out,
+			// ignoring the possibility of Case 3, which, while valid HCL encoding, is too complex to detect and handle
+			// here. Instead we will recommend users use the object encoding.
+			return nil, false, errors.WithStackTrace(MultipleBareIncludeBlocksErr{})
+		}
+
+		// Make sure this is Case 2, and not Case 3 with a single labeled block. If Case 2, update to inject the labeled
+		// version. Otherwise, return without modifying.
+		singleBlock := typed[0]
+		if jsonIsIncludeBlock(singleBlock) {
+			return updateSingleBareIncludeInParsedJSON(parsed, singleBlock)
+		}
+		return nil, false, nil
+	case map[string]interface{}:
+		if len(typed) == 0 {
+			// No include block, so don't do anything
+			return nil, false, nil
+		}
+
+		// We will only update the include block if we detect the object to represent an include block. Otherwise, the
+		// blocks are labeled so we can pass forward to the tg parser step.
+		if jsonIsIncludeBlock(typed) {
+			return updateSingleBareIncludeInParsedJSON(parsed, typed)
+		}
+		return nil, false, nil
+	}
+
+	return nil, false, errors.WithStackTrace(IncludeIsNotABlockErr{parsed: includeBlock})
+}
+
+// updateBareIncludeInParsedJSON replaces the include attribute into a block with the label "" in the json. Note that we
```

Comment:
NIT: is `""` the best name we can use here? I find it a bit hard to read, and suspect for anyone without all the context, it might be even trickier. Some ideas to think through:

1. Perhaps instead of an empty string, we pick a slightly more readable name? E.g., We could go with `"default"` if we make that name "special" and don't allow it to be used for includes names by users. Or we could go with something very unlikely to conflict, such as `"__default__"`. 
1. What if we leave the old syntax as it was: that is, `include.locals.region` would _only_ work with a single, unlabeled `include` block. If you are using labeled `include` blocks, you instead look things up via `includes["label"].locals.region`. That way, it's fully backwards compatible, and there's no awkward empty string to use as a label.

Reply:
Went with option (2): b7f61ec 

I like that because it is backward compatible.

Reply:
> NIT: Does the comment on this function need to be updated then? Could be done in a follow-up PR.

Ah on second look, the docs don't need to update because this is still true. Internally, bare include blocks are still labeled with `""`. The special logic is in the expose feature, where single bare include blocks are injected to the top level of `include` var. So the comment here is accurate, where we are still converting bare include blocks to include with `""` as label, and the special casing happens later downstream (which I commented here: https://github.com/gruntwork-io/terragrunt/blob/f02beeb7497faf7c030864b43fd56e5f5d879b04/config/cty_helpers.go#L193-L198)

Reply:
Ah, roger!